src/functions/create-session/handler.ts

```diff
@@ -70,14 +71,21 @@ const createAuthenticationSession: ValidatedEventAPIGatewayProxyEvent<any> = asy
 
   if (isBasicAuth) {
     const auth = parseBasicAuth(authHeader);
-
     if (!auth) {
       return {
         statusCode: 401,
         body: "Unauthorized"
       };
     }
 
+    const elastic = getElastic(auth);
+    if (!(await elastic.hasCurationsAccess())) {
+      return {
+        statusCode: 403,
```

Comment:
This should also return 401, as you are indeed unauthorized, when you provide invalid credentials, even when they are sent in valid structure, which is validated with the parseBasicAuth().